Nn structures nalgebra

joarndt opened the pull request:
Just added the same functionality as Lukas. Just using nalgebra and I added some dummy functions in main.rs for no warnings.

gustav1101 committed:
not working blas stuff

gustav1101 committed:
Can now create the initial NN state

joarndt committed:
solved merge conflict

joarndt committed:
put it in right folder

joarndt committed:
removed blas from Cargo.toml

joarndt committed:
removed blas and added nalgebra functionality. And dummy use of nn for no warnings

joarndt committed:
removed matrix.rs

vab9 commented:
Looks nice, maybe we should add some more documentation?

joarndt committed:
added some comments and changed nn types for nalgebra

vab9 commented on src/nn/mod.rs:
```diff
@@ -0,0 +1,102 @@
+
+        // random number between 0 and 1
+        let mut rng = rand::thread_rng();
+        let normal = Normal::new(0.0, 1.0);
```
Kann mir jemand erklaeren, warum wir hier eine Normalverteilung benutzen? Also ich weiss, dass es im "Buch" auch so steht, aber ich hab die Begruendung dafuer nicht gefunden... Hilfe @johann162 @gustav1101

vab9 committed:
improve nn/mod.rs

- removed unsafe sections
- added some comments and doc
- used StandardNormal distribution instead of Normal with standard params
- used f64 instead of f32; we can change this back if you want to
- removed unnesseary Random Trait
- minor cosmetic tweaks

vab9 committed:
use f32 instead of f64 - per dekret joarndt
